src/mqtt/mqtt_test.c

```diff
@@ -988,11 +988,6 @@ TEST( MqttTest, MQTT_Subscribe_Publish_With_Qos_1 )
             /* Timeout. */
             break;
         }
-        else if( receivedPubAck != 0 )
```

Comment:
shouldn't we still check if a puback is received to break from this loop? Wouldn't this only stop looping after the timeout or if ProcessLoop returned an error?

Reply:
See this: https://github.com/FreeRTOS/FreeRTOS-Libraries-Integration-Tests/pull/57/files#r984857650

Reply:
Then should you check for an incoming publish to break out like you do below?